server/service/src/sync/remote_data_synchroniser.rs

```diff
@@ -145,45 +127,35 @@ impl RemoteDataSynchroniser {
         batch_size: u32,
         logger: &mut SyncLogger<'a>,
     ) -> Result<(), RemotePushError> {
-        use RemotePushError as Error;
         let changelog_repo = ChangelogRepository::new(connection);
-        let change_log_filter = get_active_records_on_site_filter(connection)
-            .map_err(Error::GetActiveStoresOnSiteError)?;
+        let change_log_filter = get_active_records_on_site_filter(connection)?;
 
         loop {
             // TODO inside transaction
-            let cursor = get_push_cursor(connection).map_err(Error::DatabaseError)?;
-            let changelogs = changelog_repo
-                .changelogs(cursor, batch_size, change_log_filter.clone())
-                .map_err(Error::DatabaseError)?;
-            let change_logs_total = changelog_repo
-                .count(cursor, change_log_filter.clone())
-                .map_err(Error::DatabaseError)?;
-
-            logger
-                .progress(SyncStepProgress::Push, change_logs_total)
-                .map_err(Error::SyncLoggerError)?;
+            let cursor = get_push_cursor(connection)?;
+            let changelogs =
+                changelog_repo.changelogs(cursor, batch_size, change_log_filter.clone())?;
+            let change_logs_total = changelog_repo.count(cursor, change_log_filter.clone())?;
+
+            logger.progress(SyncStepProgress::Push, change_logs_total)?;
 
             let last_pushed_cursor = changelogs.last().map(|log| log.cursor);
 
-            let records = translate_changelogs_to_push_records(connection, changelogs)
-                .map_err(Error::TranslationError)?;
+            let records = translate_changelogs_to_push_records(connection, changelogs)?;
 
             let response = self
                 .sync_api_v5
                 .post_queued_records(change_logs_total, records)
-                .await
-                .map_err(Error::PushError)?;
+                .await?;
 
             // Update cursor only if record for that cursor has been pushed/processed
             if let Some(last_pushed_cursor_id) = last_pushed_cursor {
-                update_push_cursor(connection, last_pushed_cursor_id as u64 + 1)
-                    .map_err(Error::DatabaseError)?;
+                update_push_cursor(connection, last_pushed_cursor_id as u64 + 1)?;
             };
 
             match (response.integration_started, change_logs_total) {
                 (true, 0) => break,
-                (false, 0) => return Err(Error::IntegrationNotStarter),
+                (false, 0) => return Err(RemotePushError::IntegrationNotStarter),
```

Comment:
Should this be ??
```suggestion
                (false, 0) => return Err(RemotePushError::IntegrationNotStarted),
```

Reply:
Thanks, did this straight into head PR: bb8309272975b8f867a4d8f8c4cd2d87c73d93f2 (also changed the enum definition)